agent/internal/client/host_methods.go

```diff
@@ -279,7 +279,7 @@ func (c *hostCommunicator) GetExpansions(ctx context.Context, taskData TaskData)
 
 // Heartbeat sends a heartbeat to the API server. The server can respond with
 // an "abort" response. This function returns true if the agent should abort.
-func (c *hostCommunicator) Heartbeat(ctx context.Context, taskData TaskData) (bool, error) {
+func (c *hostCommunicator) Heartbeat(ctx context.Context, taskData TaskData) (string, error) {
```

Comment:
Does this have to return a string instead of a boolean? It seems like `true` is equivalent to `TaskFailed`, so returning an untyped string here for something that still only returns two possible values is more confusing.

Reply:
I did this since doHeartbeat returns a string, so we can directly return the result from a.comm.Heartbeat. But if the only two values are "" and evergreen.TaskFailed I'm fine with changing it back to bool